src/resolvers/profileResolver.ts

```diff
@@ -0,0 +1,53 @@
+import { Profile } from '../models/user'
+import cloudinary from 'cloudinary';
+import path from 'path';
+
+cloudinary.config({
+  cloud_name: process.env.CLOUD_NAME,
+  api_key: process.env.API_KEY,
+  api_secret: process.env.API_SECRET
+});
+
+const profileResolvers = {
+    Query: {
+        getProfile: (parent:any , args: any) => {
+            return Profile.findById(args._id)
+        }
+    },
+    Mutation: {
+        updateProfile: async (parent:any, args:any, context: any) => {
+            try {
+              let { _id, firstName, lastName, address, city, country, phoneNumber, biography, fileName } = args
+
+              const { userId } = context;
+
+                if (!userId) {
+                    throw new Error("You need to login first")
+                }
+
+                if (fileName)  {
+                  const mainDir = path.dirname(require.main?.filename)
+                  fileName = `${mainDir}/uploads/${fileName}`                
+                  const photo = await cloudinary.v2.uploader.upload(fileName)
+                }
+                
+              const updatedProfile = await Profile.findByIdAndUpdate(_id, { 
+                firstName: firstName,
+                lastName: lastName,
+                address: address,
+                city: city,
+                country: country,
+                phoneNumber: phoneNumber,
+                biography: biography,
+                avatar: `${photo?.public_id}.${photo?.format}`
```

Comment:
I believe cloudinary returns a `url` automatically from the response. To ease the integration, save the `url` to the uploaded picture. 
That way, on frontend we wont have to deal with `URL building` which is unnecessary.

```
avatar: photo.url
// the return is already a string and template literals are not needed
```